tests/test_omegaconf.py

```diff
@@ -518,3 +518,93 @@ def test_resolve(cfg: Any, expected: Any) -> None:
 def test_resolve_invalid_input() -> None:
     with raises(ValueError):
         OmegaConf.resolve("aaa")  # type: ignore
+
+
+@mark.parametrize(
+    ("cfg_params,clear_resolver_params,expected"),
+    [
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR1",
+                resolver=lambda x: str(x).lower(),
+                use_cache=False,
+                replace=True,
+            ),
+            # clear_resolver_params
+            dict(name="TR1"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR2",
+                resolver=lambda x: str(x).upper(),
+                use_cache=False,
+                replace=True,
+            ),
+            # clear_resolver_params
+            dict(name="TR2"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR3",
+                resolver=lambda x: str(x).upper(),
+                use_cache=False,
+                replace=False,
+            ),
+            # clear_resolver_params
+            dict(name="TR3"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a default resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="oc.env"),
+            # expected
+            dict(pre_addition=True, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a default resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="oc.create"),
+            # expected
+            dict(pre_addition=True, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a nonexisting resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="abcdef"),
+            # expected
+            dict(pre_addition=False, post_addition=False, post_removal=False),
+        ),
```

Comment:
To be consistent with the other tests used in this codebase, it would be good to use `param(..., id="test_name")` for each test case, e.g. see the `test_resolve` function above.
For example, the `id` for this test case can be `"nonexisting_resolver"`.

In general, please try to make the code style consistent with the other tests in this repository.

Reply:
Will do.

Reply:
Done!